roles/cpu/tasks/main.yml

```diff
@@ -71,37 +71,118 @@
 
 - name: gather cpu.count fact
   raw: cat /proc/cpuinfo | grep '^processor\s*.' | wc -l
-  register: cpu_count
+  register: cpu_count_cmd
   ignore_errors: yes
-  when: '"cpu.count" in facts_to_collect'
+  when: '"cpu.count" in facts_to_collect or "cpu.socket_count" in facts_to_collect or "cpu.core_count" in facts_to_collect'
+
+- name: extract or default cpu.count fact
+  set_fact:
+    cpu_count: "{{ cpu_count_cmd['stdout_lines'][0] | default('Not Found')}}"
+  when: '"cpu.count" in facts_to_collect or "cpu.socket_count" in facts_to_collect  or "cpu.core_count" in facts_to_collect'
 
 - name: add cpu.count to dictionary
   set_fact:
-    cpu: "{{ cpu|default({}) | combine({ item: cpu_count['stdout_lines'][0] | default('error') }) }}"
+    cpu: "{{ cpu|default({}) | combine({ item: cpu_count }) }}"
   with_items:
   - 'cpu.count'
   when: '"cpu.count" in facts_to_collect'
 
+- name: gather cpu.core_per_socket fact
+  raw: cat /proc/cpuinfo | grep '^cpu cores\s*.' | sed -n -e 's/^.*cpu cores\s*.\s*//p'
+  register: cpu_core_per_socket_cmd
+  ignore_errors: yes
+  when: '"cpu.hyperthreading" in facts_to_collect or "cpu.core_count" in facts_to_collect'
+
+- name: extract or default cpu.core_per_socket fact
+  set_fact:
+    cpu_core_per_socket: "{{ cpu_core_per_socket_cmd['stdout_lines'][0] | default('Not Found')}}"
+  when: '"cpu.hyperthreading" in facts_to_collect or "cpu.core_count" in facts_to_collect'
+
+- name: gather cpu.siblings fact
+  raw: cat /proc/cpuinfo | grep '^siblings\s*.' | sed -n -e 's/^.*siblings\s*.\s*//p'
+  register: cpu_siblings_cmd
+  ignore_errors: yes
+  when: '"cpu.hyperthreading" in facts_to_collect or "cpu.core_count" in facts_to_collect'
+
+- name: extract or default cpu.siblings fact
+  set_fact:
+    cpu_siblings: "{{ cpu_siblings_cmd['stdout_lines'][0] | default('Not Found')}}"
+  when: '"cpu.hyperthreading" in facts_to_collect or "cpu.core_count" in facts_to_collect'
+
+- name: determine cpu.hyperthreading fact
+  set_fact:
+    cpu_hyperthreading: "{{ cpu_siblings != 'Not Found' and cpu_siblings == cpu_core_per_socket }}"
```

Comment:
I think the logic on the second condition is backwards maybe? 

If hyperthreading is enabled (only for RHEL on physical hosts) the number of siblings will be 2x the number of cores. So if cores is 4 and siblings is 8 then that would mean hyperthreading is enabled but I believe this fact would be set to False.